README.md
# hrp2malaRia

Individual based malaria transmission model for the study of *P. falciparum* *pfhrp2* gene deletions. The model is an extension of the Imperial College transmission model used as part of the [Global Technical Strategy 2030 costing exercise](https://www.thelancet.com/journals/laninf/article/PIIS1473-3099(15)00423-5/fulltext), and was used to characterise the drivers of *pfhrp2* deletion selection in the following publication:

 > Watson OJ, Slater HC, Verity R, et al. Modelling the drivers of the spread of Plasmodium falciparum hrp2 gene deletions in sub-Saharan Africa. Elife 2017; 6: e25008. (https://elifesciences.org/articles/25008)

Installing *hrp2malaRia* devel
-------------
To install the development version from github (the package *devtools is required*):

```r
library(devtools)
install_github("OJWatson/hrp2malaRia")
```

Once installed, the package can be loaded using:

```r
library(hrp2malaRia)
```

Asking a question
------------------
- for bug reports, feature requests, contributions, use github's [issue system](https://github.com/OJWatson/hrp2_malaRia/issues)

## Demonstration

We can demonstrate this by simulating 2 populations of 2000 individuals. Each 
population starts with 10% frequency of hrp2 deletions. One population only
uses RDTs, whereas the other uses 70% RDTs, as well as not adhering to a
negative RDT result and still treating patients. Both populations have an 
annual EIR of 1. 

```r

# population 1
r <- hrp2_Simulation(N=2000, years=20, rdt.det = 0, EIR=1/365)

# population 2
r2 <- hrp2_Simulation(N=2000,years=20,rdt.det = .3,microscopy.use = .3, EIR = 1/365)

# and let's plot these - all veriables that begin with S. are series variables collected over time. 

# let's plot the first population
plot(r$S.Times,r$S.N.Dels, xlab = "Time (days)", ylab = "hrp2 Deletion Frequency", ylim=c(0,1), col="red", type="l", lwd=2)

# and add the second population
lines(r2$S.Times,r2$S.N.Dels,lwd=2)

# and add our legend
legend(1500, .15, legend=c("Only RDT use", "70% RDT use, and 30% nonadherence to RDT negative result"),
col=c("red", "black"), lty=1, cex=0.8,
box.lty=2, box.lwd=2)
```
![](tools/demo.png)

## Input Variables to Change

| Variable | Description                                                                   |
|----------|-------------------------------------------------------------------------------|
| *years* | Number of years simulation is run for. 20 years will be usually long enough to see hrp2 selection at low EIRs. |
| *N* | Population size. The model is stochastic and so a larger N will enable clearer patterns to be found in model outputs, however, they will take longer to run. Population of 2000 should be a good balance of speed and clarity for demonstration purposes. For the analysis in the eLife modelling paper N was set to 100,000 |
| *strains.0* | Starting ratio of strains in the population. This number must be an integer. The default is set to 10, which will mean that on initialisation there will be roughly 10 times as many wild type strains as there are hrp2 deleted. |
| *EIR* | Annual EIR, provided as a fraction, i.e. the default EIR of 20/365 represents an EIR of 20. |
| *ft* | Frequency of treatment being sought upon developing clinical disease. Must be a number between 0 and 1, with the default being 0.4, i.e. 40% of cases seek treatment. |
| *rdt.det* | The probability that an individual only infected with hrp2-deleted strains still produces a positive RDT. The default is 1, i.e. no selective pressure. For the eLife paper we used a value of 0.25 that represented the approximate 25% chance that a positive RDT would still be produced due to hrp3 protein epitope cross reactivity |
| r*dt.nonadherence* | The probability that the result of an RDT is ignored, i.e. a negative RDT due to hrp2 deletions will still be treated. Default = 0, and in the eLife paper we used a value of 10% based on data collected by the WHO. |
| *microscopy.use* | The proportion of cases that are tested by microscopy rather than RDT. Default = 0, and in the eLife paper we tested the impact of 30% of cases being diagnosed by microscopy, representing the ~70% of cases diagnosed being diagnosed by RDTs in 2014. |
| *fitness* | Comparative fitness of hrp2 deleted parasites. Default = 1, which mean the wild type is equally as likely to be passed on to mosquitoes as an hrp2 deleted parasite. 0.8 results in the hrp2 deleted strain being 80% likely to be passed on, whereas a value of 1.2 would mean it is 20% more likely. |

## Output Variables of Interest

| Variable | Description                                                                   |
|----------|-------------------------------------------------------------------------------|
| *S.Times* | The timing of when the simulated population is sampled, i.e. the time at which all other series variables (variables that begin S.) were collected |
| *S.Prev.All* | The prevalence of malaria within the whole population |
| *S.Prev.05* | The prevalence of malaria within under 5s |
| *S.N.Dels* | The proportion of all parasites that are hrp2 deleted |
| *S.N.Dels.05* | The proportion of all parasites within under 5s that are hrp2 deleted |
| *S.Incidence* | The daily clinical incidence of malaria. Multiply this by 365 to have the expected number of clinical cases of malaria a person within the whole population is expected to have in a year |
| *S.Incidence.05* | The daily clinical incidence of malaria. Multiply this by 365 to have the expected number of clinical cases of malaria a child under the age of 5 is expected to have in a year |
| *S.Prev.Mono.D* | The proportion of infected individuals who are only infected with hrp2 deleted parasites |
| *S.Prev.Mono.D.05* | The proportion of infected individuals under the age of 5 who are only infected with hrp2 deleted parasites |


NEWS.md
## hrp2malaRia 0.1.3

* Better NMF documentation

## hrp2malaRia 0.1.2

* Fix to hrp2_Simulation introduced during previous version

## hrp2malaRia 0.1.1

* microscopy 2-10 and PCR all ages now tracked in series
* Improved documentation and ability to run package not loaded

## hrp2malaRia 0.1.0

* Original release


PlottingScripts/figure1-plot.R

# figure 1 plot

# ------------------------------------------------------------------

# load packages
require(ggplot2)
require(extrafont)
require(gridExtra)
require(grid)

# Extra functions
# ------------------------------------------------------------------

fmt_dcimals <- function(decimals=0){
  function(x) format(x,nsmall = decimals,scientific = FALSE)
}

gg_color_hue <- function(n) {
  hues = seq(15, 375, length = n + 1)
  hcl(h = hues, l = 65, c = 100)[1:n]
}
# ------------------------------------------------------------------

# Read in simulation data
fig1a <- read.csv(system.file("extdata","fig1.csv",package="hrp2malaRia"),colClasses = c(rep("numeric",4),rep("NULL",6)))
colnames(fig1a)[2] <- "hrp2.Deletion.Allele.Frequency"
fig1a$Prevalence <- as.factor(fig1a$Prevalence)
fig1b <- read.csv(system.file("extdata","fig1.csv",package="hrp2malaRia"),colClasses = c(rep("numeric",1),rep("NULL",3),rep("numeric",3),rep("NULL",3)))
colnames(fig1b)[2] <- "hrp2.Deletion.Allele.Frequency"
fig1b$Starting.Frequency <- as.factor(fig1b$Starting.Frequency)
fig1cd <- read.csv(system.file("extdata","fig1.csv",package="hrp2malaRia"),colClasses = c(rep("numeric",1),rep("NULL",6),rep("numeric",3)))
colnames(fig1cd)[2] <- "hrp2.Deletion.Allele.Frequency"
# fig1a
panels1 <- ggplot(fig1a,aes(x = Years,y=hrp2.Deletion.Allele.Frequency,color=Prevalence)) + geom_smooth(span=0.2,se=F) + ylim(0,1) +
  geom_vline(xintercept = 10) + geom_vline(xintercept = 15,linetype="dashed",color="black")
panels1 <- panels1 + theme_classic() + theme_light() + theme_linedraw() + xlab("Time (years)") + ylab(expression(paste("Proportion of strains that are ",italic(pfhrp2),"-deleted")))
panels1 <- panels1 + theme(axis.title.x = element_text(margin=margin(10)), axis.title.y = element_text(margin=margin(c(0,10))),
                           legend.justification = 'left',legend.position=c(0,0.85),legend.background = element_rect(colour = 'black')) +
  scale_color_manual(labels = c("10%", "25%","60%"),values = gg_color_hue(3)) + annotate("text", x = 7.5, y =0.56125, label = "RDT Introduction")

# fig1b
panels2 <- ggplot(fig1b,aes(x = Years,y=hrp2.Deletion.Allele.Frequency,color=Starting.Frequency)) + geom_smooth(span=0.2,se=F) + ylim(0,1) +
  geom_vline(xintercept = 10) + geom_vline(xintercept = 15,linetype="dashed",color="black")
panels2 <- panels2 + theme_classic() + theme_light() + theme_linedraw() + xlab("Time (years)") + ylab(expression(paste("Proportion of strains that are ",italic(pfhrp2),"-deleted")))
panels2 <- panels2 + theme(axis.title.x = element_text(margin=margin(10)), axis.title.y = element_text(margin=margin(c(0,10))),
                           legend.justification = 'left',legend.position=c(0,0.85),legend.background = element_rect(colour = 'black')) +
  scale_color_manual(labels = c("10%", "25%","60%"),values = gg_color_hue(3)) + annotate("text", x = 7.5, y =0.56125, label = "RDT Introduction")

#fig1c
panels3l <- ggplot(fig1cd,aes(x = PCR.Prevalence,y=hrp2.Deletion.Allele.Frequency)) + geom_point(size=1,alpha=0.25)
panels3l <- panels3l + geom_smooth(se=F,method = "loess") + xlab("PfPR") + ylab(expression(paste("Proportion of strains that are ",italic(pfhrp2),"-deleted"))) + theme_classic() + theme_light() + theme_linedraw()
panels3l <- panels3l + theme(axis.title.x = element_text(margin=margin(10)), axis.title.y = element_text(margin=margin(c(0,10)))) + scale_y_continuous(labels = fmt_dcimals(2)) + ylim(0,1)
panels3l

#fig1d
panels4l <- ggplot(fig1cd,aes(x = PCR.Prevalence,y=Proportion.of.Monoinfections.Due.to.hrp2.Deleted.Strains)) + geom_point(size=1,alpha=0.25)
panels4l <- panels4l + geom_smooth(se=F,method = "loess") + xlab("PfPR") + ylab(expression(paste("Population only infected with  ",italic(pfhrp2),"-deleted mutants"))) + theme_classic() + theme_light() + theme_linedraw()
panels4l <- panels4l + theme(axis.title.x = element_text(margin=margin(10)), axis.title.y = element_text(margin=margin(c(0,10)))) + scale_y_continuous(labels = fmt_dcimals(2)) + ylim(0,1)
panels4l


## PLOTTING
# ------------------------------------------------------------------

### Letter Arrange ###

mp1 <- arrangeGrob(panels1, top = textGrob("a", x = unit(0, "npc")
                                           , y   = unit(1, "npc"), just=c("left","top"),
                                           gp=gpar(col="black", fontsize=18, fontfamily="Times New Roman")))

mp2 <- arrangeGrob(panels2, top = textGrob("b", x = unit(0, "npc")
                                           , y = unit(1, "npc"), just=c("left","top"),
                                           gp=gpar(col="black", fontsize=18, fontfamily="Times New Roman")))

mp3 <- arrangeGrob(panels3l, top = textGrob("c", x = unit(0, "npc")
                                            , y   = unit(1, "npc"), just=c("left","top"),
                                            gp=gpar(col="black", fontsize=18, fontfamily="Times New Roman")))

mp4 <- arrangeGrob(panels4l, top = textGrob("d", x = unit(0, "npc")
                                            , y = unit(1, "npc"), just=c("left","top"),
                                            gp=gpar(col="black", fontsize=18, fontfamily="Times New Roman")))
windows()
gridExtra::grid.arrange(mp1,mp2,mp3,mp4,ncol=2)



# figure 1 - supplement figure 1 plot

# ------------------------------------------------------------------

fig_s1 <- read.csv(system.file("extdata","fig1_s1.csv",package="hrp2malaRia"))
names(fig_s1) <- c("X","hrp2D","prev","Years")
panels <- ggplot(fig_s1,aes(x = Years,y=hrp2D,color=`prev`)) + geom_point()+ ylim(0,1) +
  geom_vline(xintercept = 10) + geom_vline(xintercept = 15,linetype="dashed",color="black")
panels <- panels + theme_classic() + theme_light() + theme_linedraw() + xlab("Time (years)") + ylab("PCR PfPR All Ages")
panels <- panels + theme(axis.title.x = element_text(margin=margin(10)), axis.title.y = element_text(margin=margin(c(0,10))),
                         legend.justification = 'left',legend.position=c(0,0.8),legend.background = element_rect(colour = 'black')) +
  scale_color_manual(labels = c("10%","20%","40%"),values = gg_color_hue(3),name="PCR Prevalence") +
  annotate("text", x = 7.5, y =0.56125, label = "RDT Introduction")

###


mp <- arrangeGrob(panels, top = textGrob("", x = unit(0, "npc")
                                         , y = unit(1, "npc"), just=c("left","top"),
                                         gp=gpar(col="black", fontsize=24, fontfamily="Times New Roman")))

windows()
gridExtra::grid.arrange(mp,ncol=1)



# figure 1 - supplement figure 2 plot

# ------------------------------------------------------------------

fig_s2 <- read.csv(system.file("extdata","fig1_s2.csv",package="hrp2malaRia"))
fig_s2$Comparative.Fitness <- as.factor(fig_s2$Comparative.Fitness)


panel <- ggplot(fig_s2,aes(x = Years,y=hrp2.Deletion.Allele.Frequency,color=Comparative.Fitness)) + geom_smooth(span=0.2,se=F) + ylim(0,1) +
  geom_vline(xintercept = 10) + geom_vline(xintercept = 15,linetype="dashed",color="black")
panel <- panel + theme_classic() + theme_light() + theme_linedraw() + xlab("Time (years)") + ylab(expression(paste("Proportion of strains that are ",italic(pfhrp2),"-deleted")))
panel <- panel + theme(axis.title.x = element_text(margin=margin(10)), axis.title.y = element_text(margin=margin(c(0,10))),
                           legend.justification = 'left',legend.position=c(0,0.8),legend.background = element_rect(colour = 'black')) +
  scale_color_manual(labels = c("100%","97.5","95","92.5",
                                "90%","75%","50%","25%","5%"),values = gg_color_hue(9)) +
  annotate("text", x = 7.5, y =0.56125, label = "RDT Introduction") +
  geom_segment(aes(x = 9.1, y = 0.56, xend = 9.5, yend = 0.56), colour='black', size=1,arrow = arrow(length = unit(0.2, "cm")))


###


mp <- arrangeGrob(panel, top = textGrob("", x = unit(0, "npc")
                                           , y = unit(1, "npc"), just=c("left","top"),
                                           gp=gpar(col="black", fontsize=24, fontfamily="Times New Roman")))

windows()
gridExtra::grid.arrange(mp,ncol=1)


# figure 1 - supplement figure 3 plot

# ------------------------------------------------------------------

fig_s3 <- read.csv(system.file("extdata","fig1_s3.csv",package="hrp2malaRia"))
names(fig_s3) <- c("X","hrp2D","micro","Years")

panel <- ggplot(fig_s3,aes(x = Years,y=hrp2D,color=`micro`)) + geom_smooth(span=0.2,se=F) + ylim(0,1) +
  geom_vline(xintercept = 10) + geom_vline(xintercept = 15,linetype="dashed",color="black")
panel <- panel + theme_classic() + theme_light() + theme_linedraw() + xlab("Time (years)") + ylab(expression(paste("Proportion of strains that are ",italic(pfhrp2),"-deleted")))
panel <- panel + theme(axis.title.x = element_text(margin=margin(10)), axis.title.y = element_text(margin=margin(c(0,10))),
                           legend.justification = 'left',legend.position=c(0,0.8),legend.background = element_rect(colour = 'black')) +
  scale_color_manual(labels = c("0%","30% microscopy",
                                "10% nonadherence","30% microscopy & 10% nonadherence"),values = gg_color_hue(4),
                     name="Microscopy vs Nonadherence") +
  annotate("text", x = 7.5, y =0.56125, label = "RDT Introduction")


###


mp <- arrangeGrob(panel, top = textGrob("", x = unit(0, "npc")
                                        , y = unit(1, "npc"), just=c("left","top"),
                                        gp=gpar(col="black", fontsize=24, fontfamily="Times New Roman")))

windows()
gridExtra::grid.arrange(mp,ncol=1)


# figure 1 - supplement figure 4 plot

# ------------------------------------------------------------------

fig_s4 <- read.csv(system.file("extdata","fig1_s4.csv",package="hrp2malaRia"))
names(fig_s4) <- c("X","hrp2D","nmfv","Years")

panel <- ggplot(fig_s4,aes(x = Years,y=hrp2D,color=`nmfv`)) + geom_smooth(span=0.2,se=F) + ylim(0,1) +
  geom_vline(xintercept = 10) + geom_vline(xintercept = 15,linetype="dashed",color="black")
panel <- panel + theme_classic() + theme_light() + theme_linedraw() + xlab("Time (years)") + ylab(expression(paste("Proportion of strains that are ",italic(pfhrp2),"-deleted")))
panel <- panel + theme(axis.title.x = element_text(margin=margin(10)), axis.title.y = element_text(margin=margin(c(0,10))),
                       legend.justification = 'left',legend.position=c(0,0.8),legend.background = element_rect(colour = 'black')) +
  scale_color_manual(labels = c("0%","80% nmf",
                                "100% nmf","120% nmf"),values = gg_color_hue(4),
                     name="Relative non-hrp2malaRial fever (NMF)") +
  annotate("text", x = 7.5, y =0.56125, label = "RDT Introduction")


###


mp <- arrangeGrob(panel, top = textGrob("", x = unit(0, "npc")
                                        , y = unit(1, "npc"), just=c("left","top"),
                                        gp=gpar(col="black", fontsize=24, fontfamily="Times New Roman")))

windows()
gridExtra::grid.arrange(mp,ncol=1)


# figure 1 - supplement figure 5 plot

# ------------------------------------------------------------------

fig_s5 <- read.csv(system.file("extdata","fig1_s5.csv",package="hrp2malaRia"))
names(fig_s5) <- c("X","hrp2D","level","Years")

panel <- ggplot(fig_s5,aes(x = Years,y=level,color=`nmf`)) + geom_smooth(span=0.2,se=F) + ylim(0,1) +
  geom_vline(xintercept = 10) + geom_vline(xintercept = 15,linetype="dashed",color="black")
panel <- panel + theme_classic() + theme_light() + theme_linedraw() + xlab("Time (years)") + ylab(expression(paste("Proportion of strains that are ",italic(pfhrp2),"-deleted")))
panel <- panel + theme(axis.title.x = element_text(margin=margin(10)), axis.title.y = element_text(margin=margin(c(0,10))),
                           legend.justification = 'left',legend.position=c(0,0.8),legend.background = element_rect(colour = 'black')) +
  scale_color_manual(labels = c("Main Method",
                                "92.5% Comp. Fitness, 40% Microscopy Use, 20% RDT non-adherence, 1.2x NMF Rate",
                                "95% Comp. Fitness, 30% Microscopy Use, 10% RDT non-adherence, 1x NMF Rate",
                                "97.5% Comp. Fitness, 20% Microscopy Use, 5% RDT non-adherence, 0.8x NMF Rate"),
                     values = gg_color_hue(4),
                     name="Model Assumptions") +
  annotate("text", x = 7.5, y =0.56125, label = "RDT Introduction")




###


mp <- arrangeGrob(panel, top = textGrob("", x = unit(0, "npc")
                                        , y = unit(1, "npc"), just=c("left","top"),
                                        gp=gpar(col="black", fontsize=24, fontfamily="Times New Roman")))

windows()
gridExtra::grid.arrange(mp,ncol=1)


